assets/dev/js/frontend/utils/assets-loader.js

```diff
@@ -20,27 +20,45 @@ export default class AssetsLoader {
 		const assetData = AssetsLoader.assets[ type ][ key ];
 
 		if ( ! assetData.loader ) {
-			assetData.loader = new Promise( ( resolve ) => {
-				const element = 'style' === type ? this.getStyleElement( assetData.src ) : this.getScriptElement( assetData.src );
+			assetData.loader = this.isAssetLoaded( assetData, type )
+				? Promise.resolve( true )
+				: this.loadAsset( assetData, type );
+		}
 
-				element.onload = () => resolve( true );
+		return assetData.loader;
+	}
 
-				const parent = 'head' === assetData.parent ? assetData.parent : 'body';
+	isAssetLoaded( assetData, assetType ) {
```

Comment:
It is possible that some styles or scripts have been added already by the PHP enqueue. Therefore I also check inside the DOM of these elements exist already.